app/src/main/java/unipd/se18/ocrcamera/ExtractTheText.java

```diff
@@ -0,0 +1,292 @@
+     * @author Thomas Porro(g1)
+     */
+    @Override
+    public TextRegions detectTextRegions(Bitmap image, DetectTheTextMethods method) {
+        TextAreas textContainer = new TextAreas();
+        //Put the image into a matrix, if the conversion fails it return a textContainer
+        //with the full image
+        Mat img;
+        try {
+            img = IPUtils.conversionBitmapToMat(image);
+        } catch (ConversionFailedException e){
+            Log.e(TAG, e.getErrorMessage());
+            //Creates a Point object that head to the center of the full image
+            int centerHeight = image.getHeight()/2;
+            int centerWidth = image.getWidth()/2;
+            Point center = new Point(centerWidth, centerHeight);
+
+            //Creates a Size object with the same size of the full image
+            Size imageDimensions = new Size(image.getHeight(), image.getWidth());
+
+            //Creates a RotatedRect containing the full image
+            double inclinationAngle=0;
+            RotatedRect fullImage = new RotatedRect(center, imageDimensions, inclinationAngle);
+            textContainer.addRegion(fullImage);
+            return textContainer;
+        }
+        //Do the image Processing
+        Mat filteredMat = applyFilters(img);
+        //Add each element to the TextAreas's object
+        List<RotatedRect> rectanglesList = new ArrayList<>();
+        switch(method){
+            case DETECT_MAX_TEXT_AREA: rectanglesList = detectMaxTextArea(filteredMat);
+                    break;
+            case DETECT_ALL_TEXT_AREAS: rectanglesList = detectTextAreas(filteredMat);
+                    break;
+        }
+        for(RotatedRect rectangle :  rectanglesList){
+            textContainer.addRegion(rectangle);
+        }
+        return textContainer;
+    }
+
+    /**
+     * See DetectTheTextMethods.java.
+     * @author Thomas Porro(g1)
+     */
+    @Override
+    public List<Bitmap> extractTextFromBitmap(Bitmap image, TextRegions textContainer) {
+        List<Bitmap> imgTextContainer = new ArrayList<>();
+        Mat img;
+
+        //Converts the image into a matrix
+        try {
+            img = IPUtils.conversionBitmapToMat(image);
+        } catch (ConversionFailedException e){
+            Log.e(TAG, e.getErrorMessage());
+            imgTextContainer.add(image);
+            return imgTextContainer;
+        }
+
+        /*Modifies the number of channel of the image so the  matrix is compatible  with
+          Imgproc.getRectSubPix method*/
+        Imgproc.cvtColor(img, img, Imgproc.COLOR_BGRA2BGR);
+
+        /*For each rectangle contained in textContainer extract the rectangle and saves it
+          into a bitmap*/
+        RotatedRect rectangle;
+        while(textContainer.hasNext()){
+            rectangle = (RotatedRect)textContainer.next();
+            Mat croppedMat = crop(rectangle, img);
+            //If the conversion failed return a List with only the original image
```

Comment:
In the specs it sais that if it fails it returns a list with only the full image, but doesn't it return the list of images tht it succesfully cropped + the full image?